fenix/app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -624,7 +627,7 @@ class HomeFragment : Fragment() {
 
         homeViewModel.sessionToDelete = null
 
-        updateTabCounter(requireComponents.core.store.state)
+        tabCounterView?.update(requireComponents.core.store.state)
```

Comment:
This looks pretty suspicious to me given that we just call `consumeFrom` above, which makes me wonder if this is even needed other than for possibly updating a bit sooner.

Reply:
I'm sus about this being needed, too, but I'm not blocking this PR for that investigation & potential clean-up.